tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -6,129 +6,548 @@
+            {
+                // if there is more data than expected (e.g. future version of the spec),
+                // it's should to be additive, so there must be another delimiter after `trace-tags`
+                return false;
+            }
+
+            ulong traceId;
+            ulong parentId;
+            string rawTraceId;
+            string rawSpanId;
 
 #if NETCOREAPP
-                var w3cTraceId = traceParent.AsSpan(3, 32);
-                var w3cSpanId = traceParent.AsSpan(36, 16);
-                var traceId = ParseUtility.ParseFromHexOrDefault(w3cTraceId.Slice(16));
-                if (traceId == 0)
-                {
-                    return false;
-                }
+            var w3cTraceId = header.AsSpan(start: 3, length: 32);
+            var w3cSpanId = header.AsSpan(start: 36, length: 16);
+            traceId = ParseUtility.ParseFromHexOrDefault(w3cTraceId[16..]);
+
+            if (traceId == 0)
+            {
+                return false;
+            }
+
+            parentId = ParseUtility.ParseFromHexOrDefault(w3cSpanId);
+
+            if (parentId == 0)
+            {
+                return false;
+            }
 
-                var parentId = ParseUtility.ParseFromHexOrDefault(w3cSpanId);
+            rawTraceId = w3cTraceId.ToString();
+            rawSpanId = w3cSpanId.ToString();
 
-                spanContext = new SpanContext(traceId, parentId, samplingPriority, serviceName: null, null, w3cTraceId.ToString(), w3cSpanId.ToString());
+            var traceFlags = header.AsSpan(start: 53, length: 2);
+            var sampled = (ParseUtility.ParseFromHexOrDefault(traceFlags) & 1) == 1;
 #else
-                var w3cTraceId = traceParent.Substring(3, 32);
-                var w3cSpanId = traceParent.Substring(36, 16);
-                var traceId = ParseUtility.ParseFromHexOrDefault(w3cTraceId.Substring(16));
-                if (traceId == 0)
+            rawTraceId = header.Substring(startIndex: 3, length: 32);
+            rawSpanId = header.Substring(startIndex: 36, length: 16);
+            traceId = ParseUtility.ParseFromHexOrDefault(rawTraceId.Substring(16));
+
+            if (traceId == 0)
+            {
+                return false;
+            }
+
+            parentId = ParseUtility.ParseFromHexOrDefault(rawSpanId);
+
+            if (parentId == 0)
+            {
+                return false;
+            }
+
+            var traceFlags = header.Substring(53, 2);
+            var sampled = (ParseUtility.ParseFromHexOrDefault(traceFlags) & 1) == 1;
+#endif
+
+            traceParent = new W3CTraceParent(
+                traceId: traceId,
+                parentId: parentId,
+                sampled: sampled,
+                rawTraceId: rawTraceId,
+                rawParentId: rawSpanId);
+
+            return true;
+        }
+
+        internal static bool TryParseTraceState(string header, out W3CTraceState traceState)
```

Comment:
This needs to parse the value of `dd` and ignore other key/value pairs (but persistent and propagate them).

Reply:
done